src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java

```diff
@@ -0,0 +1,260 @@
+import net.snowflake.client.jdbc.telemetry.TelemetryData;
+import net.snowflake.client.jdbc.telemetry.TelemetryField;
+import net.snowflake.client.jdbc.telemetryOOB.TelemetryEvent;
+import net.snowflake.client.jdbc.telemetryOOB.TelemetryService;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * @author mknister
+ *
+ * This SnowflakeSQLLoggedException class extends the SnowflakeSQLException class to add OOB telemetry data for sql
+ * exceptions. Not all sql exceptions require OOB telemetry logging so the exceptions in this class should only be
+ * thrown if there is a need for logging the exception with OOB telemetry.
+ */
+
+public class SnowflakeSQLLoggedException extends SnowflakeSQLException
+{
+  /*
+  OOB telemetry instance
+   */
+  private TelemetryService oobInstance = TelemetryService.getInstance();
+
+  /*
+  IB telemtry instance
+   */
+  private Telemetry ibInstance = null;
+
+  private final static ObjectMapper mapper =
+          ObjectMapperFactory.getObjectMapper();
+
+  /**
+   * Function to create a TelemetryEvent log from the JSONObject and exception and send it via OOB telemetry
+   *
+   * @param value JSONnode containing relevant information specific to the exception constructor that
+   *              should be included in the telemetry data, such as sqlState or vendorCode
+   * @param ex    The exception being thrown
+   */
+  private void sendOutOfBandTelemetryMessage(JSONObject value, SnowflakeSQLLoggedException ex)
+  {
+    TelemetryEvent.LogBuilder logBuilder = new TelemetryEvent.LogBuilder();
+    StringWriter sw = new StringWriter();
+    PrintWriter pw = new PrintWriter(sw);
+    ex.printStackTrace(pw);
+    String stackTrace = sw.toString();
+    value.put("Stacktrace", stackTrace);
+    TelemetryEvent log = logBuilder
+        .withName("Exception: " + ex.getClass().getSimpleName())
+        .withValue(value)
+        .build();
+    oobInstance.report(log);
+  }
+
+  /**
+   * Function to create a TelemetryClient log and send it via in-band telemetry
+   *
+   * @param value ObjectNode containing information specific to the exception constructor that should be included in
+   *              the telemetry log, such as SQLState or reason for the error
+   * @param ex The exception being thrown
+   * @return true if in-band telemetry log sent successfully or false if it did not
+   */
+  private void sendInBandTelemetryMessage(ObjectNode value, SnowflakeSQLLoggedException ex) throws IOException {
+    StringWriter sw = new StringWriter();
+    PrintWriter pw = new PrintWriter(sw);
+    ex.printStackTrace(pw);
+    String stackTrace = sw.toString();
+    value.put("Stacktrace", stackTrace);
+    value.put("Exception", ex.getClass().getSimpleName());
+    ibInstance.addLogToBatch(new TelemetryData(value, System.currentTimeMillis()));
+    ibInstance.sendBatch();
```

Comment:
You don't have to use and expose `sendBatch` right? You still can use `sendBatchAsync` and call `future.get()` and return the boolean back so that the caller can use it to decide whether to send OOB.